readthedocs/doc_builder/exceptions.py

```diff
@@ -57,6 +57,11 @@ class BuildUserError(BuildBaseException):
         "Ensure your project is configured to use the output path "
         "'$READTHEDOCS_OUTPUT/html' instead."
     )
+    NO_CONFIG_FILE_DEPRECATED = gettext_noop(
+        "Not using a '.readthedocs.yaml' configuration file is deprecated. "
+        "Add a configuration file to your project to make it build successfully. "
+        "Read more at https://docs.readthedocs.io/en/stable/config-file/v2.html"
```

Comment:
I just reviewed the text from the notification on your other PR, I'd copy that here. It feels like these should all be the same text.

Reply:
I don't think we can copy the exact same text here since this is an error and the other is a warning. At this point builds are already failing. However, I updated the copy to follow with a similar idea.

Reply:
Aye, we can't copy paste, but but I think the wording makes more sense here:

https://github.com/readthedocs/readthedocs.org/pull/10354/files#diff-b1aca2d9afa30600f245fd743c9345f2d605956fd94b6c76559852d0e610e8fdR172-R173

That is, we tell the user the outcome to them, not what we did to our code.

I can't suggest anything here for some reason, but what I'm describing is:

```
        "A configuration file is required in order to build your project. "
        "Add a configuration file to your project to make it build successfully. "
        "Read more at https://docs.readthedocs.io/en/stable/config-file/v2.html"
```

Reply:
I think it's because I was looking at a single commit, because the commit history makes review hard on this PR. Guessing it just needs rebased.